rmf_fleet_adapter/src/rmf_fleet_adapter/agv/RobotUpdateHandle.cpp

```diff
@@ -828,6 +828,35 @@ void RobotUpdateHandle::reassign_dispatched_tasks()
   }
 }
 
+//==============================================================================
+RobotUpdateHandle::LiftDestination::LiftDestination()
+  : _pimpl(rmf_utils::make_impl<Implementation>())
+{
+  // Do nothing
+}
+
+//==============================================================================
+const std::string& RobotUpdateHandle::LiftDestination::lift() const
+{
+  return _pimpl->lift;
+}
+
+//==============================================================================
+const std::string& RobotUpdateHandle::LiftDestination::level() const
+{
+  return _pimpl->level;
+}
+
+//==============================================================================
+std::optional<RobotUpdateHandle::LiftDestination>
+RobotUpdateHandle::lift_destination() const
+{
+  if (const auto context = _pimpl->get_context())
+  {
+    return context->final_lift_destination();
+  }
```

Comment:
shouldn't there be a return when there is no context?

Reply:
Great catch, fixed here: [ca7d865](https://github.com/open-rmf/rmf_ros2/pull/376/commits/ca7d86593b3ae9c5f899fab4dc2103861a080a38)

I'm very annoyed that the C++ compiler doesn't prevent this by default 😮‍💨